j5/backends/hardware/sr/v4/motor_board.py

```diff
@@ -0,0 +1,190 @@
+        ports = list(filter(lambda x: x.manufacturer == "Student Robotics", ports))
+        ports = list(filter(lambda x: x.product == "MCV4B", ports))
+
+        # Get a list of boards from the ports.
+        boards: List[Board] = []
+        for port in ports:
+            boards.append(
+                MotorBoard(
+                    port.serial_number,
+                    SRV4MotorBoardHardwareBackend(port.device, serial_class),
+                ),
+            )
+
+        return boards
+
+    @handle_serial_error
+    def __init__(self, serial_port: str, serial_class: Type[Serial] = Serial) -> None:
+        # Initialise our stored values for the state.
+        self._state: List[MotorState] = [
+            MotorSpecialState.BRAKE
+            for _ in range(0, 2)
+        ]
+
+        self._serial = serial_class(
+            port=serial_port,
+            baudrate=1000000,
+            timeout=0.25,
+        )
+
+        # Check we have the correct firmware version.
+        version = self.get_firmware_version()
+        if version != "3":
+            raise CommunicationError(
+                f"Unexpected firmware version: {version}, expected: 3.",
+            )
+
+        # Brake both of the motors
+        for i, val in enumerate(self._state):
+            self.set_motor_state(i, val)
+
+    def __del__(self) -> None:
+        """Clean up device on destruction of object."""
+        # Brake both of the motors for safety
+        for i, val in enumerate(self._state):
+            self.set_motor_state(i, MotorSpecialState.BRAKE)
+
+        self._serial.flush()
+        self._serial.close()
+
+    @handle_serial_error
+    def send_command(self, command: int, data: Optional[int] = None) -> None:
+        """Send a serial command to the board."""
+        command_bytes = chr(command).encode('utf-8')
+        bytes_written = self._serial.write(command_bytes)
+        if len(command_bytes) != bytes_written:
+            raise CommunicationError(
+                "Mismatch in command bytes written to serial interface.",
+            )
+
+        if data is not None:
+            data_bytes = chr(data).encode()
+            bytes_written = self._serial.write(data_bytes)
+            # It is not possible to test the following if statement without refactor.
+            if len(data_bytes) != bytes_written:  # pragma: nocover
+                raise CommunicationError(
+                    "Mismatch in data bytes written to serial interface.",
+                )
+
+    @handle_serial_error
+    def read_serial_line(self) -> str:
+        """Read a line from the serial interface."""
+        bdata = self._serial.readline()
+
+        if len(bdata) == 0:
+            raise CommunicationError(
+                "Unable to communicate with motor board. ",
+                "Is it correctly powered?",
+            )
+
+        ldata = bdata.decode('utf-8')
+        return ldata.rstrip()
+
+    def get_firmware_version(self) -> Optional[str]:
```

Comment:
AFAIK in python we don't typically _aim_ for having `get_` or `set_`, instead using `@property` here with just `firmware_version` might be more pythonic?

Reply:
So this is part of a separate issue that bugged me whilst writing this. I have written the code here to match the existing style and opened #236 to track the overall issue.